### PR TITLE
Re-enable Kotlin DSL tests and mark them `@LeaksFileHandles`

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

File: subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/AbstractTestDirectoryProvider.java [hunk 1]
```diff
@@ -99,16 +94,32 @@ void doCall() throws IOException {
...
+                    System.err.println(cleanupErrorMessage());
                     e.printStackTrace(System.err);
                 } else {
-                    throw new GradleException(message, e);
+                    throw new GradleException(cleanupErrorMessage(), e);
                 }
             }
         }
+
+        private boolean suppressCleanupErrors() {
```
Comment (wolfs) [inline subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/AbstractTestDirectoryProvider.java]: I would have called these methods `isSuppressCleanupErrors()`, `getTestClass()`, etc., so the actual code could have been the same (i.e. could reference to the getters as if they were fields).

Comment (bamboo) [reply subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/AbstractTestDirectoryProvider.java]: Thanks, @wolfs. Interestingly enough I avoided the `get` prefix for that exact reason, that is, to keep the code somewhat similar to the original based on fields.

> i.e. could reference to the getters as if they were fields

This one is in Java.

**Comment by bamboo** [comment]
Thanks for the review!

File: subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/AbstractTestDirectoryProvider.java (hunk 1 above)
Comment (wolfs) [reply subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/AbstractTestDirectoryProvider.java]: Ahhh, Java .... I knew I missed something.

**Pull request merged**